client/app/src/accounts/account.controller.js

```diff
@@ -1523,6 +1528,93 @@
         }
       };
 
+      function upload() {
+        var options = {
+          title: "Upload Image",
+          filters: [
+            { name: 'Images', extensions: ['jpg', 'png', 'gif'] }
+          ],
+          properties: ["openFile"],
+        };
+        var userPath = 'assets/images/user/'
+        var dirPath = path.resolve(__dirname, userPath);
+
+        require('electron').remote.dialog.showOpenDialog(options, function(fileName) {
+          if (fileName === undefined) return;
+          fileName = fileName[0]
+
+          var baseName = path.basename(fileName);
+          var newFileName = path.join(dirPath, baseName);
+
+          var readStream = fs.createReadStream(fileName);
+          readStream.on("error", function(err) {
+            console.log("Error Reading File: " + err);
+            $mdToast.show(
+              $mdToast.simple()
+              .textContent(gettextCatalog.getString('Error Adding Background.'))
+              .hideDelay(3000)
+            );
+            return;
+          });
+
+          var writeStream = fs.createWriteStream(newFileName);
+          writeStream.on("error", function(err) {
+            console.log("Error Writing File: " + err);
+            $mdToast.show(
+              $mdToast.simple()
+              .textContent(gettextCatalog.getString('Error Adding Background.'))
+              .hideDelay(3000)
+            );
+            return;
+          });
+          writeStream.on("close", function(ex) {
+            $mdToast.show(
```

Comment:
Probably a good idea to pull the toast logic into its own method where you pass in the text to be displayed

Reply:
Thanks for the feedback. I went ahead and made a showToast function. Now the toasts are all made with either showToast or formatAndToastError. I also removed my unnecessary log statements. Thanks!

Reply:
there are already a few Toast function, but it is not rational:
- https://github.com/ArkEcosystem/ark-desktop/blob/master/client/app/src/accounts/account.controller.js#L278
- https://github.com/ArkEcosystem/ark-desktop/blob/master/client/app/src/accounts/account.controller.js#L338

Would be nice to revamp the toast management into a single identified service instead

Reply:
would be for another PR